File: daily/Linked List/86-Partition-list.py
class Solution:
    def partition(self, head: Optional[ListNode], x: int) -> Optional[ListNode]:
        before_dummy = ListNode(0)
        after_dummy = ListNode(0)

        before = before_dummy
        after = after_dummy

        curr = head
        while curr:
            if curr.val < x:
                before.next = curr
                before = before.next
            else:
                after.next = curr
                after = after.next
            curr = curr.next

        after.next = None
        before.next = after_dummy.next
        return before_dummy.next    

File: daily/Linked List/86-Partition-list.go
package daily

func partition(head *ListNode, x int) *ListNode {
	beforeDummy := &ListNode{0, nil}
	afterDummy := &ListNode{0, nil}

	before := beforeDummy
	after := afterDummy

	curr := head
	for curr != nil {
		if curr.Val < x {
			before.Next = curr
			before = before.Next
		} else {
			after.Next = curr
			after = after.Next
		}
		curr = curr.Next
	}

	after.Next = nil
	before.Next = afterDummy.Next
	return beforeDummy.Next
}
